zeppelin-web/src/components/navbar/navbar.html

```diff
@@ -86,7 +86,7 @@
             <i ng-if="!navbar.connected" class="fa fa-circle server-disconnected"
                uib-tooltip="WebSocket Disconnected" tooltip-placement="bottom" style="margin-top: 7px; vertical-align: top"></i>
             <button ng-if="ticket" class="nav-btn dropdown-toggle" type="button" data-toggle="dropdown" style="margin:11px 5px 0 0; padding-left: 0px;">
-              <span class="username">{{ticket.principal}}</span>
+              <span class="username">{{ticket.screenUsername}}</span>
```

Comment:
I am also looking into Pac4J and enabling OAuth for Zeppelin.  I was working on my own pull request when I found this one.  The Pac4jPrincipal implements java.security.Principal.  I think it is cleaner to modify the SecurityUtils.getPrincipal() method to check if the object implements the Principal interface like so...
```
public static String getPrincipal() {
    if (!isEnabled) {
      return ANONYMOUS;
    }
    Subject subject = org.apache.shiro.SecurityUtils.getSubject();
    String principal;
    if (subject.isAuthenticated()) {
      Object principalObj = subject.getPrincipal();
      if(principalObj instanceof Principal) {
        principal = ((Principal) principalObj).getName();
      } else {
        principal = subject.getPrincipal().toString();
      }
    } else {
      principal = ANONYMOUS;
    }
    return principal;
  }
```

Then we don't have to change the navbar.html and navbar.controller.js